src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -133,6 +138,51 @@ private int CalculateGetMoreProtocolNumberToReturn()
             return numberToReturn;
         }
 
+        /// <summary>
+        /// Closes cursor.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public void Close(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
In most .NET classes similar to this one calling Close and Dispose are considered equivalent (except that Dispose is sync-only and won't throw exceptions).

So Close and CloseAsync need to call Dispose also.

You'll need some mechanism to avoid infinite recursion (Close calls Dispose which calls Close etc...).

Reply:
I think this method should be:

    public void Close(CancellationToken cancellationToken = default(CancellationToken))
    {
        CloseCursor(cancellationToken);
        Dispose();
     }

If CloseCursor and Dispose are designed to be called multiple times then Close can be called multiple times also.